build/codegen/codegen.sh

```diff
@@ -14,29 +14,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# outputs absolute path for given relative path.
+function toAbsolutePath(){
+  echo "$(cd "$(dirname "$1")"; pwd)/$(basename "$1")"
+}
+
+
 GROUP_VERSIONS="ceph.rook.io:v1"
 
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
+output_base="$(dirname "${BASH_SOURCE[0]}")/../../../../.."
+output_pkg="github.com/rook/rook/pkg/client"
+apis_pkg="github.com/rook/rook/pkg/apis"
+
+# check that output directory for generated packages already exists.
+if [ ! -d "$output_base/$apis_pkg" ]; then
+  echo "apis pkg $(toAbsolutePath $output_base/$apis_pkg) does not exist."
+  echo "please make sure that project located in 'github.com/rook/rook' directory"
```

Comment:
I worry about requiring this for everyone, it may be too strict, if it causes all builds to fail. The codegen is really an optional step for local developer builds, we just want the CI to fail if it hasn't been successfully run. Can we make it so this only fails the `make codegen` step, but not `make build`?